### PR TITLE
KK-1069 | Update to HDS v3.4

### PR DESCRIPTION
**Pull request by karisal-anders**
## Description

Current API & UIs are incompatible because of recent changes to API unrelated to this PR,
and because of that testing environment and review environments for UIs are broken until
the incompatibilities are fixed.

This means to test this PR you have to spin up a compatible local [kukkuu API v.1.17.2](https://github.com/City-of-Helsinki/kukkuu/releases/tag/kukkuu-v1.17.2) and use it against this PR.

### Summary
 - Update to HDS v3.4
 - Update to RHHC v1.0.0-alpha250 which uses HDS v3.4
 - Use HDS v3 Footer directly
 - Use HDS v3 Header through RHHC
 - Update snapshots & fix tests

#### 1. **Footer** contents are **not configurable** from Headless CMS but are hardcoded like they were before

image

#### 2. Header menu's pages' videos' horizontal sizes bounce from width to width

Open any page from the header menu, e.g.
 - "Kulttuurikummit"
 - or "Dhaqanka iyo mataaneynta caruurta"
 - or "Hälsningar från evenemangen"
 
and resize the window horizontally from full screen width to
narrower and narrower and see how the contents change.

Example:
 - URL: http://localhost:3000/fi/tutustu-toimintaan/

 - 
 - 
 - 
 - 
☝️ Video is full width suddenly!
 - 

#### 3. Editing guardian's info opens modal which focuses (should not) on languages spoken at home dropdown

edit-button
edit-opened

## Context

[KK-1069](https://helsinkisolutionoffice.atlassian.net/browse/KK-1069)

#### Frontpage
frontpage-fi

#### Header before logging in
image

#### Header after logging in (wide)
image

#### Header after logging in (medium)
image

#### Header after logging in (small)
image

#### Footer (wide)
image

#### Footer (small)
image

#### Edit child's information modal window
image

#### Frontpage
frontpage-sv

#### Header before logging in
image

#### Header after logging in (wide)
image

#### Header after logging in (medium)
image

#### Header after logging in (small)
image

#### Footer (wide)
image

#### Footer (small)
image

#### Edit child's information modal window
image

#### Frontpage
frontpage-en

#### Header before logging in
image

#### Header after logging in (wide)
image

#### Header after logging in (medium)
image

#### Header after logging in (small)
image

#### Footer (wide)
image

#### Footer (small)
image

#### Edit child's information modal window
image

[KK-1069]: https://helsinkisolutionoffice.atlassian.net/browse/KK-1069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr519.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr519.dev.hel.ninja :crying_cat_face::anger::boom::boom:

File: src/assets/styles/layout.scss [hunk 1]
```diff
@@ -1,21 +1,15 @@
...
-  md: 992px,
-  lg: 1200px,
-  xlg: 1600px,
+  xs: $breakpoint-xs,
+  s: $breakpoint-s,
+  m: $breakpoint-m,
+  l: $breakpoint-l,
+  xl-minus: $container-width-xl,
+  xl: $breakpoint-xl,
+  xxl: 1600px, // $container-width-xl * 4 / 3
```
Comment (nikomakela) [inline src/assets/styles/layout.scss:10]: if there is no support for this in HDS, do we need to support that width at all? 1600px is quite much.

Comment (nikomakela) [reply src/assets/styles/layout.scss:10]: In events-monorepo there is 
```
$breakpoints: (
  xs: $breakpoint-xs,
  s: $breakpoint-s,
  m: $breakpoint-m,
  l: $breakpoint-l,
  xl: $breakpoint-xl,
  // xxl => 1568px
  xxl: $breakpoint-xl + $breakpoint-xs,
);
```

**Review by nikomakela — COMMENTED** [review]
I reviewed all the code changes and to me they looked ok. I will still take a look how the app really renders.

**Comment by nikomakela** [comment]
> 2. Header menu's pages' videos' horizontal sizes bounce from width to width
Open any page from the header menu, e.g.
>
>"Kulttuurikummit"
or "Dhaqanka iyo mataaneynta caruurta"
or "Hälsningar från evenemangen"
and resize the window horizontally from full screen width to
narrower and narrower and see how the contents change.

The youtube video figure becomes full width, because the PageContentLayout changes the flex-direction from "column" to "row". I don't think this is a bad thing -- it's part of the current layout design and the result is not even bad.

**Review by nikomakela — APPROVED** [review]
I would just merge this so we could test it more in staging. The listed caveats have been there earlier too or are actually arguably some new unpolished features, but I would not say that they are breaking anything or blocking the deployment.

I don't personally like the new profile icon in the header at all -- I think it's bad design -- but it is something that HDS is possibly going to improve quite soon.

Good job!

**Pull request merged**